### PR TITLE
Non transient value type error

### PR DESCRIPTION
**Pull request by AlonTalmi**
If you register a value type as singleton or scoped you get a compile error.

The compile error happens because there's a nullable field saved for each registration, which in reference type is implicitly castable to the implementation type, but in value type you have to cast it manually or use the Value property.

The 1st commit is a Unit Test showcasing the problem

### CODE REVIEW

File: src/Jab.FunctionalTests.Common/ContainerTests.cs [hunk 1]
```diff
@@ -1272,6 +1272,21 @@ public void SupportsNamedServices()
         [Singleton(typeof(IAnotherService), typeof(AnotherServiceImplementation), Name="OnlyNamed")]
         [Singleton(typeof(ServiceImplementationWithNamed<IService>))]
         internal partial class SupportsNamedServicesContainer
+		{
```
Comment (pakrym) [inline src/Jab.FunctionalTests.Common/ContainerTests.cs:1275]: Something is wrong with the formatting here. 

Do you mind replacing these tabs with spaces?

File: src/Jab/ContainerGenerator.cs [hunk 2]
```diff
@@ -38,7 +38,14 @@ private void GenerateCallSiteWithCache(CodeWriter codeWriter, string rootReferen
                     });
             }
 
-            valueCallback(codeWriter, w => w.Append($"{cacheLocation}"));
+            if (serviceCallSite.ImplementationType.IsValueType)
+            {
+                valueCallback(codeWriter, w => w.Append($"{cacheLocation}!.Value"));
```
Comment (pakrym) [inline src/Jab/ContainerGenerator.cs:43]: Is `!` required? There is an `if ... == null` check right above.

Comment (AlonTalmi) [reply src/Jab/ContainerGenerator.cs:43]: Technically, but not really, it's a == default check which still causes the compiler to throw an error, I can change it to == null and remove the !

Comment (pakrym) [reply src/Jab/ContainerGenerator.cs:43]: `== null` sounds good.

**Comment by pakrym** [comment]
Thank you!

**Pull request merged**